botocore/stub.py

```diff
@@ -362,3 +388,13 @@ def _validate_response(self, operation_name, service_response):
             raise ParamValidationError(
                 report=(
                     "Service response should only contain ResponseMetadata."))
+
+
+class StubRequest(object):
+    """
+    A tracked request that was made to a stubbed client. Has two attributes,
+    ``operation_name`` and ``params``.
+    """
+    def __init__(self, operation_name, params):
+        self.operation_name = operation_name
```

Comment:
There's no such thing as a read-only attribute in python, just magic to mask it. I could do `_operation_name` proxied by an `@property` function that returns `self._operation_name` but then one can still modify `_operation_name`...